src/Symfony/Contracts/Service/LocaleAwareInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\Service;
+
+/**
+ * LocaleAwareInterface should be implemented by classes that needs the current locale.
+ *
+ * @author Pierre Bobiet <pierrebobiet@gmail.com>
+ */
+interface LocaleAwareInterface
+{
+    /**
+     * Sets the current locale.
+     *
+     * @param string $locale
+     */
+    public function setLocale($locale);
```

Comment:
Right now, I'm :-1: to add a setter in a contract.
That's a pure marker interface that encourages building mutable services.
We should seek for another design IMHO.